bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_types.go

```diff
@@ -94,6 +97,32 @@ type KubeadmConfigSpec struct {
 	// For more information, refer to https://github.com/kubernetes-sigs/cluster-api/pull/2763#discussion_r397306055.
 	// +optional
 	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
+
+	// Ignition contains Ignition specific configuration.
+	//
+	// +optional
+	Ignition *IgnitionSpec `json:"ignition,omitempty"`
+}
+
+// IgnitionSpec contains Ignition specific configuration.
+type IgnitionSpec struct {
+	// ContainerLinuxConfig contains CLC specific configuration.
+	//
+	// +optional
+	ContainerLinuxConfig *ContainerLinuxConfig `json:"containerLinuxConfig,omitempty"`
+}
+
+// ContainerLinuxConfig contains CLC specific configuration.
+//
+// We use structured type here to keep space for adding additional fields, for example 'version'.
+type ContainerLinuxConfig struct {
+	// AdditionalConfig contains additional configuration which will be added bootstrap data in CLC format.
```

Comment:
```suggestion
	// AdditionalConfig contains additional configuration which will be appended/merged [1] to the Ingnition configuration generated by the bootstrapper controller.
       // [1] https://coreos.github.io/ignition/operator-notes/#config-merging
```

link thanks to @bgilbert

Reply:
Done, please have a look again :)